### PR TITLE
MBS-8041 Refresh expired OpenID token

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
We support Open Id tokens in k8s for authentication and authorization. 
`kubectl` refreshes tokens automatically and stores changes to kube config file.

[Kubernetes - OpenID Connect Tokens](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens)

How they look in the config:

```yaml
users:
- name: username
  user:
    as-user-extra: {}
    auth-provider:
      config:
        client-id: k8s
        client-secret: xxx
        extra-scopes: groups
        id-token: xxx                       <-------
        idp-issuer-url: https://xxx
        refresh-token: xxx
      name: oidc
```

`id-token` is the  [Open id token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) in Jwt format:

```json
{
 "iss":"https://xxx",
 "sub":"xxx",
 "aud":"k8s",
 "exp":1585929826,  <---
 "iat":000,
 "at_hash":"xxx",
 "email":"user@mail",
  
```
 `exp` - is the expiration date. This is how the client knows it in advance.

Unfortunately, fabric8 kubernetes-client doesn't refresh tokens.
I've reused implementation from the official client (https://github.com/kubernetes-client/java/pull/871)

Known issues:
- The client persists yaml correctly but with another formatting style. I think we can live with it for now.

### CODE REVIEW

File: subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/Kubernetes.kt [hunk 1]
```diff
@@ -32,14 +35,27 @@ fun createKubernetesClient(
...
                 }
 
-                // work with multiple namespaces/contexts not supported
+                // working with multiple namespaces/contexts is not supported
                 require(getNamespace() == namespace) {
                     "kubernetes.context.namespace should be $namespace. " +
                         "Namespace hardcoded in plugin, and this check only prevents from using wrong context"
                 }
+                requestConfig.oauthTokenProvider = oauthTokenProvider(configFile)
             }
```
Comment (eugene-krivobokov) [inline subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/Kubernetes.kt:44]: https://github.com/fabric8io/kubernetes-client/issues/2111

File: subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/KubernetesCredentials.kt [hunk 2]
```diff
@@ -20,6 +20,7 @@ sealed class KubernetesCredentials : Serializable {
     ) : KubernetesCredentials(), Serializable
 }
 
+// TODO: get rid of this default. autoConfig is enabled by default
```
Comment (eugene-krivobokov) [inline subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/KubernetesCredentials.kt:23]: This refactoring is out of scope

**Review by sergioboi — APPROVED** [review]

**Review by dsvoronin — APPROVED** [review]

File: subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/Kubernetes.kt (hunk 1 above)
Comment (eugene-krivobokov) [reply subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/Kubernetes.kt:44]: Fixed
